specify url in accessory

jaseclamp opened the pull request:
Hey mate just some updates to the accessory because we wanted to be able to specify urls to clear also (when we just want to clear /css/styles.css for example). If blank old behavior applies. 

added gzip to curl - most browsers request gzip so the hash in varnish includes gzip. By sending purge request without gzip it likely won't clear the gzipped versions in cache. 
in accessory allow specify url or leave blank to clear all.
accessory now reports back varnish response OR curl error for better feedback.

kevincupp commented on third_party/purge/views/accessory_purge_varnish.php:
```diff
@@ -1,21 +1,38 @@
+			var url = $('#purge_url').val(); 
+			
+			if( url != '')
+			if( ! url.match(/^\/[^\/]/) ) { $('#purge_url').val('URL must start with one "/"'); return; }
```
This seems more suited for an alert, that way it doesn't wipe out the URL in the text box. How about making this an alert as well?

kevincupp replied:
Then you could probably remove that click event above. I guess I'm just thinking of the case where you enter something like "about", and then it warns you to add a slash, then you just have to add a slash to the beginning instead of typing out the whole thing again.

jaseclamp commented:
ok, I've adjusted that. we also encountered a situation where we needed to add msm awareness to the url patterns mcp so I've added that to this also. thanks for accepting my contributions :)

kevincupp commented:
Thanks! Looks good. Just a couple more things:

- [ ] Since you're using the `site_id` column now, we need to add it back to the module install routine. Don't worry about adding it to the update routine, it'll probably be fine.
- [ ] Add these changes to the changelog and up the version number. Maybe call it 1.1.1.

jaseclamp commented:
There ya go. Thanks again.

kevincupp commented on third_party/purge/acc.purge.php:
```diff
@@ -63,7 +66,17 @@ public function process_purge_request()
+				
+				echo $resp; //depending on curl to send varnish's response or an error
+				
+				die; //we die here instead of return to cut of template debugging 
```
Hmm hang on a second. Since you die inside the loop, this no longer supports purging multiple servers specified via the `varnish_site_url` config. It'll only do one. Is there another way to handle this?

jaseclamp commented:
That is a good point! I'll have a think about that.

jaseclamp commented:
OK, just adjusted so it collects all responses and relays those back.

kevincupp commented:
Thanks, looks good!
